### PR TITLE
Set v5 as default java version

### PR DESCRIPTION
**Pull request by cschubertcs**
This PR sets the default version of the Java SDK to v5 and moves v4 over to the versioned directories.

### CODE REVIEW

File: sidebarsDocsJava.js [hunk 1]
```diff
@@ -23,15 +22,7 @@ module.exports = {
           label: 'OpenAPI',
           items: [
             'features/rest/overview',
-            'features/rest/generate-rest-client',
-            {
-              type: 'category',
-              label: 'Released Clients',
-              items: [
-                'features/rest/clients/scp-workflow-rest-api',
-                'features/rest/clients/btp-business-rules-rest-api'
```
Comment (CharlesDuboisSAP) [inline sidebarsDocsJava.js:32]: Bookmarks on those 2 links and 'guides/4.0-upgrade' will be dead. They need to be redirected to /v4/...

Comment (cschubertcs) [reply sidebarsDocsJava.js:32]: I tried now multiple variants in the redirection, and it seems to have problems with the `.` in the path.
I didn't have time to deeper investigate, but a quick google search didn't help.
With that I would drop it, at least for the moment.

File: docs-java/getting-started.mdx [hunk 2]
```diff
@@ -63,51 +62,48 @@ After providing all the interactive values to the CLI it will generate your firs
...
+groupId: foo.bar.cloud.app
 artifactId: mydreamapp
 version: 1.0-SNAPSHOT
 package: foo.bar.cloud.app
-gitignore: .gitignore
-skipUsageAnalytics: false
  Y: :
 [INFO] ----------------------------------------------------------------------------
-[INFO] Using following parameters for creating project from Archetype: scp-cf-spring:RELEASE
+[INFO] Using following parameters for creating project from Archetype: scp-cf-spring-jakarta:RELEASE
```
[the remark this answers is not in the document]
Comment (cschubertcs) [reply docs-java/getting-started.mdx:89]: Hm, I tried to use our archetype, but it fails with `The desired archetype does not exist (com.sap.cloud.sdk.archetypes:spring-boot3:5.0.0-alpha-SNAPSHOT)` or `The desired archetype does not exist (com.sap.cloud.sdk.archetypes:spring-boot3:RELEASE)`

Could someone have a look here? Maybe I'm just using the wrong settings.xml, but cannot go deeper into this right now.

[the remark this answers is not in the document]
Comment (CharlesDuboisSAP) [reply docs-java/getting-started.mdx:89]: Removed the `RELEASE` option generates the `5.0.0-SNAPSHOT` archetype. Everything else fails, I even get this error:

**Review by jjtang1985 — APPROVED** [review]
LGTM.
As this PR focuses on the switch of the default SDK version.
I would leave the rest of the review comments to the follow up PRs.
The default version needs to be switched for:
- google index
- v5 links shared in Blog Post, OSPO reports etc

**Pull request merged**